Agents/APIAgent/README.md

```diff
@@ -0,0 +1,163 @@
+
+##2.2 Agent Operation
+Upon an API is registered to the API Agent, the Agent will conduct two actions:
+1. Immediately create a Timeseries if there is no existing one associated to the target data IRI. Immediately download the data from the API and instantiate or update the Timeseries Instance.
+2. Register a periodical scheduled job to repeat action 1.
+
+To register an API, one should use the derivation client functions to create a derivation through the API agent, an example is shown below: 
+```bash
+from pyderivationagent import PyDerivationClient
+from pyderivationagent.data_model import ONTODERIVATION_DERIVATIONWITHTIMESERIES
+derivation_client = PyDerivationClient(triples_base_url, kg_endpoint, kg_endpoint)
+derivation_client.createSyncDerivationForNewInfo(API_AGENT_IRI, [IRI_of_API_Mapping],
+                                                                  ONTODERIVATION_DERIVATIONWITHTIMESERIES)
+```
+The full time-sequence of API Agent actions is illustrated as the below diagram. Note that the only required action for the user is to register the API via two steps: 1) Insert proper meta triples into KG. 2) Register the API via creating a derivation.
+Once this is done, the API Agent automatically manages the instantiation and all follow-up updates of the API data. User can modify the meta-data at any time (e.g, the external source has changed to a different URL), the change will be reflected after the next scheduled update.
+
+<p align="center">
+    <img src="./imgs/sequenceAPIAGENTv2.png" alt="drawing" width="800"/>
+</p>
+
+##2.2 Extra Options
+###2.2.1 API with dynamically generated URL
+It is common for external data API to use dynamically generated access URL. For this purpose, we define a new rdf property in `RMLTS.OWL`: `rmlts:hasDynamicTarget`. Find in [test_triples] folder an example to define an API source with dynamically generated URL: [api_temperature.ttl].
+ 
+###2.2.2 API-to-TS mapping with Algebra
+Some source-data-to-target-Timeseries mapping may need algebra operations. For examples, the raw data contains 4 set of sub-category quantities and the target quantity is the sum of them. Such operations can be declared as a Named Individual of type `rmlts:Algebra`. And a corresponding implementation should be put in the [calculations module]. We have implemented one example: `rmlts:Sum`. An API meta-data example can be found in [test_triples] folder: [api_pvcapacity.ttl].
+
+
+
+
+#3. Using the Agent
+
+A local running container can be initiated via:
+```
+docker compose -f "docker-compose-test-dockerised.yml" up -d --build
```

Comment:
The fact that it's "dockerised" goes without saying, given that it's a docker compose file. To call it "test" is confusing though, as this is normally reserved for unit/regression/integration tests. I suspect that this is meant to be used not in that sense, so perhaps it should be called `docker-compose.yml`?

Reply:
Changed to 'docker-compose'.